src/powerplaylists/utils.py

```diff
@@ -120,6 +124,23 @@ def get_or_default(key: str, default: Any) -> Any:
         self.daemon_pidfile = get_or_default("daemon_pidfile", self.daemon_pidfile)
         self.verify_mode = VerifyMode[get_or_default("verify_mode", self.verify_mode.name).upper()]
 
+    def __ensure_directories_exist(self) -> None:
+        """Ensure all required directories exist, creating them if necessary."""
+        # Create the main app directory first (e.g., ~/.power-playlists)
+        pathlib.Path(Constants.APP_HOMEDIR).mkdir(parents=True, exist_ok=True)
+
+        # Create the user config directory (e.g., ~/.power-playlists/userconf)
+        pathlib.Path(self.user_config_dir).mkdir(parents=True, exist_ok=True)
+
+        # Create the cache directory (e.g., ~/.power-playlists/cache)
+        pathlib.Path(self.cache_dir).mkdir(parents=True, exist_ok=True)
+
+        # Ensure the parent directory for the log file exists
+        pathlib.Path(self.log_file_path).parent.mkdir(parents=True, exist_ok=True)
+
+        # Ensure the parent directory for the daemon PID file exists
+        pathlib.Path(self.daemon_pidfile).parent.mkdir(parents=True, exist_ok=True)
+
```

Comment:
- remove all the comments
- turn this into a loop to have better DRY
- inline this function